src/folder/folder.model.ts

```diff
@@ -0,0 +1,34 @@
+import { ServerError } from '../errors/application';
+import { Schema, model, Mongoose } from 'mongoose';
+import { IFolder } from './folder.interface';
+
+export const folderSchema = new Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    owner: {
+      type: String,
+      required: true,
+    },
+    parent: {
+      type: String,
+      required: true,
+    },
+    files: {
+      type: [String],
+      required: true,
+    },
+    folders: {
```

Comment:
Should be a virtual field.
Read about it [here](http://thecodebarbarian.com/mongoose-virtual-populate)